core/drivers/imx_wdog.c

```diff
@@ -124,21 +104,19 @@ static TEE_Result imx_wdog_base(vaddr_t *wdog_vbase)
 	}
 
 	/* search the first usable wdog */
-	for (i = 0; i < ARRAY_SIZE(wdog_path); i++) {
-		off = fdt_path_offset(fdt, wdog_path[i]);
+	while (true) {
+		off = fdt_node_offset_by_compatible(fdt, off, "fsl,imx21-wdt");
 		if (off < 0)
-			continue;
+			return TEE_ERROR_ITEM_NOT_FOUND;
 
 		st = _fdt_get_status(fdt, off);
 		if (st & DT_STATUS_OK_SEC)
 			break;
 	}
 
-	if (i == ARRAY_SIZE(wdog_path))
+	if (!off)
 		return TEE_ERROR_ITEM_NOT_FOUND;
```

Comment:
These two lines can be removed, in case no further node is found the loop already returns `TEE_ERROR_ITEM_NOT_FOUND`.